File: euler_020.java
import java.math.BigInteger;

public class euler_020
{
  static BigInteger factorial(int number)
  {
    BigInteger product = BigInteger.valueOf(1);
    for (int i = 1; i <= number; i++)
      product = product.multiply(BigInteger.valueOf(i));
    return product;
  }

  public static void main(String[] args)
  {
    int[] targets = { 10, 100 };
    for (int target : targets)
    {
      int sum = 0;
      // take the factorial of target number, the factorial being a BigInteger,
      // convert the BigInteger into a String, then convert the String into an
      // array of digit-characters, before summing up all the digits in the array
      BigInteger factorial = factorial(target);
      for (char digit : factorial.toString().toCharArray())
        sum += digit - '0';
      System.out.println("factorial(" + target + "): " + factorial + ", sum: " + sum);
    }
  }
}

File: euler_020.py
def factorial(number):
  product = 1
  for i in range(1, number+1):
    product *= i
  return product

targets = [ 10, 100 ]
for target in targets:
  # take the factorial of target number, the factorial being a huge integer
  fact = factorial(target)
  # convert the huge integer into a string, break that string into an array of
  # characters, turn that array into an array of digits, then take the sum of
  # all the digits in the array of digits
  total = sum(map(int, list(str(fact))))
  print("factorial(%d): %d, sum: %d" %(target, fact, total))
